specutils/wcs/wcs_adapter.py

```diff
@@ -0,0 +1,107 @@
+import abc
+import logging
+from collections import namedtuple
+
+import six
+
+WCSAxes = namedtuple('WCSAxes', ['longitude', 'latitude', 'cubeface',
+                                 'spectral', 'stokes', 'celestial', ])
+
+
+class WCSAdapterMeta(type):
+    """
+    Meta class to add any subclasses to a registry associating some WCS object
+    with a specific ~`WCSAdapter` class that `specutils` can interface with.
+    """
+    def __init__(cls, name, bases, dct):
+        if not hasattr(cls, 'registry'):
+            cls.registry = {}
+        else:
+            wcs_class = dct.get('wrapped_class', None)
+
+            if wcs_class is not None:
+                logging.info("Added {} to adapter registry.".format(wcs_class))
+                cls.registry[wcs_class] = cls
+
+        super(WCSAdapterMeta, cls).__init__(name, bases, dct)
+
+
+@six.add_metaclass(type('WCSAdapterMetaProxy',
```

Comment:
Given comments below on py2 support, you probably don't need this.

Also, you could go Python 3.6 only and drop the whole metaclass (astropy/astropy#7144)

Reply:
Yes, definitely. I suppose the question is whether to be python 3+ only, or python 3.6+ only?

Reply:
Go 3.6+ people should upgrade 😆